doc/source/whatsnew/v1.3.0.rst

```diff
@@ -283,6 +283,7 @@ Indexing
 - Bug in :meth:`DataFrame.loc` dropping levels of :class:`MultiIndex` when :class:`DataFrame` used as input has only one row (:issue:`10521`)
 - Bug in setting ``timedelta64`` values into numeric :class:`Series` failing to cast to object dtype (:issue:`39086`)
 - Bug in setting :class:`Interval` values into a :class:`Series` or :class:`DataFrame` with mismatched :class:`IntervalDtype` incorrectly casting the new values to the existing dtype (:issue:`39120`)
+- Bug in incorrectly raising when setting a new column that cannot be held in the existing ``frame.columns`` instead of casting to a compatible dtype (:issue:`39068`)
```

Comment:
hmm isn't this directly `Index.insert` ?

Reply:
Yes, could flesh this out (.reset_index is also affected)

Reply:
yeah i think its worth being more explict here as this is a non-trivial change